pertpy/tools/_perturbation_space/_perturbation_space.py

```diff
@@ -1,107 +1,20 @@
-    This class also provides various methods to evaluate perturbation spaces
-    and to perform arithmetics with it.
-
-    Currently available methods:
-    # TODO all others with lots of detail
-    - "pseudobulk": Pseudobulk based.
-        Determines the pseudobulks of the conditions.
-
-    Attributes:
-        method: Name of perturbation space function.
-
-    Examples:
-        >>> import pertpy as pt
-
-        >>> adata = pt.dt.papalexi_2021()["rna"]
-        >>> pbps = pt.tl.PerturbationSpace(method='pseudobulk')
-        >>> perturbation_space = pbps()
-        >>> TODO add something here
     """
 
-    def __init__(
-        self,
-        method: Literal["differential", "centroid", "pseudobulk"] = "pseudobulk",
-    ):
-        """Initialize PerturbationSpace class.
-
-        Args:
-            adata: The AnnData object to calculate the perturbation space for.
-            method: PerturbationSpace method to use. Defaults to 'pseudobulk'.
-        """
-        if method == "differential":
-            from pertpy.tools._perturbation_space._simple import DifferentialSpace
-
-            method_fct = DifferentialSpace()
-        elif method == "centroid":
-            from pertpy.tools._perturbation_space._simple import CentroidSpace
-
-            method_fct = CentroidSpace()  # type: ignore
-        elif method == "pseudobulk":
-            from pertpy.tools._perturbation_space._simple import PseudobulkSpace
-
-            method_fct = PseudobulkSpace()  # type: ignore
-        else:
-            raise ValueError(f"Method {method} not recognized.")
-        self.method_fct = method_fct
-
-    def __call__(
-        self,
-        adata: AnnData,
-        *,
-        reference_key: str = "control",
-        target_col: str = "perturbations",
-        layer_key: str = None,
-        new_layer_key: str = None,
-        embedding_key: str = None,
-        new_embedding_key: str = None,
-        copy: bool = False,
-        **kwargs,
-    ) -> AnnData:
-        """Compute perturbation space.
-
-        Args:
-            adata: The AnnData object to compute the perturbation space for.
-            reference_key: The reference sample. Defaults to 'control'.
-            target_col: The column containing the sample states (perturbations). Defaults to 'perturbations'.
-            layer_key: The layer of which to use the transcription values for to determine the perturbation space.
-            new_layer_key: The name of the new layer to add to the AnnData object.
-            embedding_key: The obsm matrix of which to use the embedding values for to determine the perturbation space.
-            new_embedding_key: The name of the new obsm key to add to the AnnData object.
-
-        Returns:
-            The determined perturbation space inside an AnnData object.
-            # TODO add more detail
-
-        Examples:
-            >>> import pertpy as pt
+    @abstractmethod
+    def __call__(self, *args, **kwargs):
```

Comment:
This could eventually have a more well-defined set of parameters that all implementations should have at least - see my general comment.

Reply:
now it's changed and each method has its own space